### PR TITLE
Bring ALPS ODLS up to par regarding wdir.

### LINKED ISSUE #3296: --wdir ignored in recent master
`--wdir` is no longer an effective option, e.g.,:

```
mkdir tmp
mpirun -np 1 -wdir $PWD/tmp pwd
```
returns `$PWD` and not `$PWD/tmp`.

I noticed some related commits (https://github.com/open-mpi/ompi/pull/3217) recently, could these be the cause?

### PR DESCRIPTION
**Pull request by marksantcroos**
This fixes #3296 which was caused by #3217.

Signed-off-by: Mark Santcroos <mark.santcroos@rutgers.edu>

### CODE REVIEW

File: orte/mca/odls/alps/odls_alps_module.c [hunk 1]
```diff
@@ -423,6 +423,11 @@ static int do_child(orte_odls_spawn_caddy_t *cd, int write_fd)
     sigprocmask(0, 0, &sigs);
     sigprocmask(SIG_UNBLOCK, &sigs, 0);
 
+    /* take us to the correct wdir */
+    if (NULL != cd->wdir) {
+        chdir(cd->wdir);
```
Comment (hppritcha) [inline orte/mca/odls/alps/odls_alps_module.c]: I'd like to see this handle the case that chdir returns with error.

Comment (marksantcroos) [inline orte/mca/odls/alps/odls_alps_module.c]: I don't disagree strongly, although I think your request is already covered by the call to orte_util_check_context_cwd() (https://github.com/open-mpi/ompi/blob/master/orte/util/context_fns.c#L56).

Note that my addition is mirroring the existing code in the default ODSL module  (https://github.com/open-mpi/ompi/blob/master/orte/mca/odls/default/odls_default_module.c#L417).

**Comment by marksantcroos** [comment]
@rhc54 Would you like me to add the check in both places? Happy to do so.

**Comment by rhc54** [comment]
Oh - yes please! Much appreciated 😄

**Comment by hppritcha** [comment]
I'm okay with improving possible error return from ```chdir``` with some other PR.

**Comment by rhc54** [comment]
Very nice - thanks!!!

**Comment by marksantcroos** [comment]
> Very nice - thanks!!!

Hold your guns! ;)

It turned out to be less straight forward than I anticipated ...

Just noticed that basename is not set in the Cray case.
Should I possibly hardcode that to "orted" anyway?

**Comment by rhc54** [comment]
I think that should be fine

**Comment by marksantcroos** [comment]
Ok, thanks for the feedback. Tested on Titan and MacOS, its good to go for me.

**Pull request merged**